File: backend/src/main/java/com/team05/petmeeting/domain/user/service/UserService.java
package com.team05.petmeeting.domain.user.service;

import com.team05.petmeeting.domain.user.entity.User;
import com.team05.petmeeting.domain.user.errorCode.UserErrorCode;
import com.team05.petmeeting.domain.user.repository.UserRepository;
import com.team05.petmeeting.global.exception.BusinessException;
import jakarta.transaction.Transactional;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
@Transactional
public class UserService {
    private final UserRepository userRepository;

    public User findById(Long id) {
        return userRepository.findById(id).orElseThrow(() -> new BusinessException(UserErrorCode.USER_NOT_FOUND));
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/user/service/UserService.kt
package com.team05.petmeeting.domain.user.service

import com.team05.petmeeting.domain.user.entity.User
import com.team05.petmeeting.domain.user.errorCode.UserErrorCode
import com.team05.petmeeting.domain.user.repository.UserRepository
import com.team05.petmeeting.global.exception.BusinessException
import jakarta.transaction.Transactional
import org.springframework.stereotype.Service

@Service
@Transactional
class UserService(
    private val userRepository: UserRepository
) {
    // Todo : 레포지토리 전환 후 elvis 로 throw
    fun findById(id: Long): User = userRepository.findById(id)
        .orElseThrow { BusinessException(UserErrorCode.USER_NOT_FOUND) }
}
